compiler/src/dmd/dsymbol.d

```diff
@@ -269,6 +269,12 @@ extern (C++) class Dsymbol : ASTNode
         loc = Loc(null, 0, 0);
     }
 
+    final extern (D) this(const ref Loc loc) nothrow
```

Comment:
I don't understand, why is this needed? Can't you just set the loc in the constructor in `attrib.d`?

Reply:
No, because `loc` field is declared `const`, so it can be set only by a constructor
https://github.com/dlang/dmd/blob/master/compiler/src/dmd/dsymbol.d#L258

Reply:
Which you are declaring in StorageClassDeclaration.

Reply:
> I don't understand, why is this needed? Can't you just set the loc in the constructor in attrib.d?

**in the constructor in attrib.d**